File: 884. Uncommon Words from Two Sentences/uncommonFromSentences.py
from typing import List

class Solution:
  def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:
    words = s1.split() + s2.split()
    wordCount = {}
    uncommonWords = []

    for word in words:
      if word not in wordCount:
        wordCount[word] = 1
      else:
        wordCount[word] += 1

    for word in wordCount:
      if wordCount[word] == 1:
        uncommonWords.append(word)

    return uncommonWords
  

s1 = "this apple is sweet"
s2 = "this  apple is sour"
solution = Solution()
print("uncommonFromSentences: ", solution.uncommonFromSentences(s1, s2)) # ["sweet", "sour"]

File: 884. Uncommon Words from Two Sentences/uncommonFromSentences.js
/**
 * @param {string} s1
 * @param {string} s2
 * @return {string[]}
 */
const uncommonFromSentences = (s1, s2) => {
  const words = [...s1.split(' '), ...s2.split(' ')];
  const wordsCount = {};

  for (const word of words) {
    wordsCount[word] = wordsCount[word] + 1 || 1;
  }

  return Object.keys(wordsCount).filter(word => wordsCount[word] === 1);
};

const s1 = "this apple is sweet";
const s2 = "this apple is sour";
console.log("uncommonFromSentences: ", uncommonFromSentences(s1, s2)); // ["sweet","sour"]
